File: backend/app/schemas/self_assessment.py
import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class SelfAssessmentCreate(BaseModel):
    date: datetime.date
    performed_well_partner: bool
    note: str | None = None


class SelfAssessmentResponse(BaseModel):
    id: UUID
    date: datetime.date
    performed_well_partner: bool
    note: str | None
    integrity_score: int
    created_at: datetime.datetime
    updated_at: datetime.datetime

    model_config = ConfigDict(from_attributes=True)


class SelfAssessmentHistoryResponse(BaseModel):
    entries: list[SelfAssessmentResponse]
    average_score: float

File: frontend/types/self-assessment.ts
export interface SelfAssessmentResponse {
  id: string
  date: string
  performed_well_partner: boolean
  note: string | null
  integrity_score: number
  created_at: string
  updated_at: string
}

export interface SelfAssessmentHistoryResponse {
  entries: SelfAssessmentResponse[]
  average_score: number
}

export interface SelfAssessmentCreateInput {
  date: string
  performed_well_partner: boolean
  note?: string
}
